src/componentRenderer.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { renderToString } from 'react-dom/server'
+import { uniqueId } from 'lodash'
+
+import {
+  ServerStyleSheet,
+  __DO_NOT_USE_OR_YOU_WILL_BE_HAUNTED_BY_SPOOKY_GHOSTS
+} from 'styled-components'
+
+// For Jest. See: https://github.com/styled-components/styled-components/issues/1692
+const { StyleSheet } = __DO_NOT_USE_OR_YOU_WILL_BE_HAUNTED_BY_SPOOKY_GHOSTS
```

Comment:
LOLOL, what's the tldr here?

Reply:
The TLDR is that I don't know why they won't expose the `StyleSheet` instance! (Or 🍺 😄) When the SSR round occurs it builds up a style queue and they reallly don't want people messing with it. But unfortunately lots of tooling (like Jest) builds upon `jsdom` and when styled-components mounts it thinks it's on the client (because of the existence of `global.window`). This provides a super weird escape hatch....